src/lib/transit/planning/path-finder.ts

```
@@ -0,0 +1,242 @@
+ * Generic priority queue implementation for path finding algorithms
+ */
+export class PriorityQueue<T> {
+  private items: { element: T; priority: number }[] = [];
+  private keyMap: Map<string, number> = new Map(); // Maps keys to their indexes
+
+  constructor(private getKey: (element: T) => string) {}
+
+  push(element: T, priority: number): boolean {
+    const key = this.getKey(element);
+    const existingIndex = this.keyMap.has(key)
+      ? this.items.findIndex((item, i) => this.getKey(item.element) === key)
```

ESLint

[error] 'i' is defined but never used.

(@typescript-eslint/no-unused-vars)